strings/hamming_distance.py

```diff
@@ -0,0 +1,36 @@
+def hamming_distance(string1: str, string2: str) -> int:
+    """Calculate the Hamming distance between two equal length strings
+    In information theory, the Hamming distance between two strings of equal
+    length is the number of positions at which the corresponding symbols are
+    different. https://en.wikipedia.org/wiki/Hamming_distance
+
+    Args:
+        string1 (str): Sequence 1
+        string2 (str): Sequence 2
+
+    Returns:
+        int: Hamming distance
+
+    >>> hamming_distance("python", "python")
+    0
+    >>> hamming_distance("karolin", "kathrin")
+    3
+    >>> hamming_distance("00000", "11111")
+    5
+    """
+    assert len(string1) == len(string2)
```

Comment:
How about raising an exception if they're not equal, and add a test case to it?

Reply:
Yes, I added an exception. Thank you!